hasUniqueChars.c

```diff
@@ -68,48 +66,33 @@ bool hasUniqueChars(char * inputStr) {
   // if a bit at a position is 1, then we have seen that character
   unsigned long checkBitsA_z = 0;   // for checking A through z and {|}~
   unsigned long checkBitsexcl_amp =0;  // for checking ! though @ 
+  unsigned long empty_except_idx =0;
 
   char nextChar;         // next character in string to check
 
-  // -------------------------------------------------------------
-  // This section contains code to display the initial values of checkBitsA_z
-  // and checkBitsexcl_amp, for debugging purposes. 
-  // It also illustrates how to use the seeBits function for debugging.
-  // Printed values should initially be all zeros
-  // TODO: remove or comment out this code when satisfied of function correctness
-  
-  char debug_str_A_z[128];
-  strcpy(debug_str_A_z, "checkBitsA_z before: \n");
-  seeBits(checkBitsA_z, debug_str_A_z);
-  
-  char debug_str_excl_amp[128];
-  strcpy(debug_str_excl_amp, "checkBitsexcl_amp before: \n");
-  seeBits(checkBitsexcl_amp, debug_str_excl_amp);
-  // -------------------------------------------------------------
-
-  // TODO: Declare additional variables you need here
-
   
   for(i = 0; i < strlen(inputStr); i++) {
     nextChar = inputStr[i];
-    // TODO: Add your code here to check nextChar, see if it is a duplicate, and update the checkBits variables
-
-    // -------------------------------------------------------------
-    // Below this are examples of debugging print statements you could use
-    // Move/use as makes sense for you!
-    // Modify to work on checkBitsexcl_amp
-    // TODO: Comment out or remove when your function works correctly
-    printf("nextchar int value: %d\n", nextChar);
-    char char_str[2] = "\0";
-    char_str[0] = nextChar;
-    strcpy(debug_str_A_z, "nextchar: ");
-    strcat(debug_str_A_z, char_str);
-    strcat(debug_str_A_z,", checkBitsA_z: \n");
-    seeBits(checkBitsA_z, debug_str_A_z);
-    // ------------------------------------------------------------- 
+    if(nextChar >= 65){
+      //Creating a new int with a 1 at index given by the value of nextChar and 0s elsewhere
+      int mask = 1 << nextChar;
+      //If both the checkBitsA_z and our new int have a 1 at the same place, this boolean is true, and therefore we have found a duplicate
+      if(checkBitsA_z & mask){
+        fprintf(stderr, "There is a duplicate character! It is: %c and input is: %s\n", nextChar,inputStr);
+        return false;
+        exit(EXIT_FAILURE);
```

Comment:
No, just return false, no need for the exit (and, after the return, it will never execute)